include/fixed_containers/enum_map.hpp

```diff
@@ -499,8 +499,22 @@ class EnumMapBase
     template <class... Args>
     constexpr std::pair<iterator, bool> emplace(Args&&... args) noexcept
     {
-        std::pair<K, V> as_pair{std::forward<Args>(args)...};
```

Comment:
im curious, how does `std::pair<K, V> as_pair{std::forward<Args>(args)...};` work if the parameter pack has sizeof...(Args) != 2 values?

Reply:
This raises a good point. Since `std::map` stores `std::pair`, the emplace target is a pair.
Hence, the original code was creating a `std::pair` and then using that, so possible compilation issues would be handled by constructing the `std::pair`. 

This would reject parameter packs that were not compatible (including `sizeof...(Args) != 2 ` although you can actually have different count). However, this is problematic for non-moveable/non-copyable types due to the temporary pair.

I have revised the code and tests to be more explicit about what is supported.